File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/UpdateBigSegmentContextTargetsExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class UpdateBigSegmentContextTargetsExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var included = new SegmentUserList(
            add: [
            ],
            remove: [
            ]
        );

        var excluded = new SegmentUserList(
            add: [
            ],
            remove: [
            ]
        );

        var segmentUserState = new SegmentUserState(
            included: included,
            excluded: excluded
        );

        try
        {
            new SegmentsApi(config).UpdateBigSegmentContextTargets(
                projectKey: null,
                environmentKey: null,
                segmentKey: null,
                segmentUserState: segmentUserState
            );
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling SegmentsApi#UpdateBigSegmentContextTargets: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-node/src/UpdateBigSegmentContextTargetsExample.ts
import * as fs from 'fs';
import api from "launchdarkly_client"
import models from "launchdarkly_client"

const apiCaller = new api.SegmentsApi();
apiCaller.setApiKey(api.SegmentsApiApiKeys.ApiKey, "YOUR_API_KEY");

const included: models.SegmentUserList = {
  add: [
  ],
  remove: [
  ],
};

const excluded: models.SegmentUserList = {
  add: [
  ],
  remove: [
  ],
};

const segmentUserState: models.SegmentUserState = {
  included: included,
  excluded: excluded,
};

apiCaller.updateBigSegmentContextTargets(
  "projectKey_string", // projectKey
  "environmentKey_string", // environmentKey
  "segmentKey_string", // segmentKey
  segmentUserState,
).catch(error => {
  console.log("Exception when calling SegmentsApi#updateBigSegmentContextTargets:");
  console.log(error.body);
});
